docs/Planning_The_Custom_Device.md

```diff
@@ -0,0 +1,225 @@
+
+![](images/Extra_Page_Names_Array.jpg)<br />
+**Figure: Extra Page Names Array**
+
+If you do not use the Custom Device Template Tool to create extra pages, you must manually add and configure them.
+
+Manually adding extra pages to a custom device after running the Custom Device Template Tool is cumbersome. Avoid this issue by creating a few extra pages beyond what you think will be necessary. Unused extra pages are not executed, but they do consume marginal space on disk.
+
+### GUID
+
+When you associate an extra page with a channel or section, you override the default page for that item. This is done by specifying a GUID when the item is created, or by setting the item’s GUID using **[NI VeriStand](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_ni_veristand_vis_pal/)** » **[Custom Device API](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_custom_device_api_vis_pal/)** » **[Configuration VIs](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_configuration_vis_pal/)** » **[Item Properties](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_item_properties_vis_pal/)** » **[Set Item GUID](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_set_item_guid_vi/)**.
+
+![](images/NI_VeriStand_API_to_Set_an_Item_GUID.JPG)
+**Figure: NI VeriStand API to Set an Item’s GUID**
+
+The Custom Device Template Tool generates a GUID for each extra page in the Extra Page Names (No Extension) control.
+There is a GUID Generator VI in <vi.lib>\NI Veristand\Custom Device Tools\Custom Device Template Tool\Custom Device Template Tool.lvlib:GUID Generator.vi. Before you can run this VI by itself, you must change the Custom Device Template Tool.lvlib\subVIs access scope to public, and set the
+ 
+VI’s Execution Priority to Normal. There are also a variety of free GUID generators on-line.
+XML Declaration
+The custom device API associates a channel or section with a GUID. The custom device XML associates the GUID with the page VI. The page and its GUID must be declared in the custom device XML <PAGES> section within a <PAGE> schema. If the developer planned for the extra pages before running the Custom Device Template Tool, the tool makes the appropriate entries in the custom device XML file for each extra page.
+
+![](images/Custom_Device_XML.JPG)
+**Custom Device XML Showing the Page Name, GUID, and VI**
+
+### Build Specification
+
+Extra pages are dynamically called VIs. Since they are not a part of the custom device’s VI hierarchy, they must be explicitly included in the custom device's Build Specification. If the developer planned for the required extra pages before running the Custom Device Template Tool, the tool configures the build specifications to include the extra pages into the initialization library.
```

Comment:
```suggestion
Extra pages are dynamically called VIs. Since they are not a part of the custom device’s VI hierarchy, they must be explicitly included in the custom device's Build Specification. If the developer planned for the required extra pages before running the niveristand-custom-device-wizard, the wizard configures the build specifications to include the extra pages into the initialization library.

```